cmd/servicecmd/servicestartcmd.go

```diff
@@ -254,7 +254,14 @@ func StartReplicatorService() {
 	dClient := dconfigclient.NewDconfigClient(cfg.GetServiceConfig(serviceName), serviceName)
 	sCommon := common.NewService(serviceName, uuid.New(), cfg.GetServiceConfig(serviceName), common.NewUUIDResolver(meta), hwInfoReader, reporter, dClient, common.NewBypassAuthManager())
 
-	h, tc := replicator.NewReplicator(serviceName, sCommon, meta, replicator.NewReplicatorClientFactory(cfg, common.GetDefaultLogger()), cfg)
+	allHosts := cfg.GetReplicatorConfig().GetReplicatorHosts()
+	allSplitHosts := make(map[string][]string)
+	for deployment, hosts := range allHosts {
```

Comment:
These logic is more like a util function rather than some logic that is specific to replicator, so I slightly feel better to keep it outside.